plugin/main/src/kotlinx/benchmark/gradle/internal/generator/workers/GenerateJsSourceWorker.kt

```diff
@@ -0,0 +1,85 @@
+package kotlinx.benchmark.gradle.internal.generator.workers
+
+import kotlinx.benchmark.gradle.KlibResolver
+import kotlinx.benchmark.gradle.Platform
+import kotlinx.benchmark.gradle.SuiteSourceGenerator
+import kotlinx.benchmark.gradle.createModuleDescriptor
+import kotlinx.benchmark.gradle.internal.generator.RequiresKotlinCompilerEmbeddable
+import org.gradle.api.file.*
+import org.gradle.api.provider.*
+import org.gradle.workers.WorkAction
+import org.gradle.workers.WorkParameters
+import org.jetbrains.kotlin.descriptors.ModuleDescriptor
+import org.jetbrains.kotlin.storage.LockBasedStorageManager
+import org.jetbrains.kotlin.storage.StorageManager
+import java.io.File
+
+/**
+ * Generates JavaScript benchmarking source code.
+ *
+ * This worker requires `kotlin-compiler-embeddable` and *must* be run in an isolated classpath.
+ *
+ * @see kotlinx.benchmark.gradle.JsSourceGeneratorTask
+ */
+@RequiresKotlinCompilerEmbeddable
+internal abstract class GenerateJsSourceWorker : WorkAction<GenerateJsSourceWorker.Params> {
```

Comment:
This and `GenerateWasmSourceWorker` are almost identical. The only exceptions are `useBenchmarkJs` param and a platform supplied to `SuiteSourceGenerator`.

Perhaps, we can use a single set of classes (worker and parameters) instead?

Reply:
> I would like to resolve this, and simplify the code, but can we agree to do it in later PR in order to keep the changes here focused? I can make an issue.

Sure, fixing it later would be fine

Reply:
https://github.com/Kotlin/kotlinx-benchmark/issues/240